pcsx2/Elfheader.cpp

```diff
@@ -62,17 +62,27 @@ void ElfObject::initElfHeaders(bool isPSXElf)
 
 	DevCon.WriteLn( L"Initializing Elf: %d bytes", data.GetSizeInBytes());
 
-	if ( header.e_phnum > 0 )
-		proghead = (ELF_PHR*)&data[header.e_phoff];
+	if (header.e_phnum > 0)
+	{
+		if ((header.e_phoff + sizeof(ELF_PHR)) <= data.GetSizeInBytes())
```

Comment:
Hmm, a few new warnings on these, would be nice to take care of them.

https://github.com/PCSX2/pcsx2/runs/6128211125?check_suite_focus=true#step:11:503